cpp/src/io/cudftable.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025, NVIDIA CORPORATION.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "cudf/detail/nvtx/ranges.hpp"
+
+#include <cudf/contiguous_split.hpp>
+#include <cudf/detail/copy.hpp>
+#include <cudf/detail/utilities/vector_factories.hpp>
+#include <cudf/io/data_sink.hpp>
+#include <cudf/io/datasource.hpp>
+#include <cudf/io/experimental/cudftable.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/utilities/error.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_buffer.hpp>
+
+#include <cstring>
+#include <vector>
+
+namespace cudf::io::experimental {
+namespace detail {
+
+namespace {
+
+/**
+ * @brief Simple binary file format header for CUDFTable
+ *
+ * The CUDFTable format stores a table in a simple binary layout:
+ * - Magic number (4 bytes): "CDFT"
+ * - Version (4 bytes): uint32_t format version (currently 1)
+ * - Metadata length (8 bytes): uint64_t size of the metadata buffer in bytes
+ * - Data length (8 bytes): uint64_t size of the data buffer in bytes
+ * - Metadata (variable): serialized column metadata from pack()
+ * - Data (variable): contiguous device data from pack()
+ */
+struct cudftable_header {
+  static constexpr uint32_t magic_number = 0x54464443;  ///< "CDFT" in little-endian
```

Comment:
We still have time to update the magic number if needed? How about `CUDF` ?

Reply:
What does CDFT stand for?

Reply:
changed to CTBL